app/models/concerns/alegre_v2.rb

```diff
@@ -231,14 +231,24 @@ def parse_similarity_results(project_media, field, results, relationship_type)
       }.reject{ |k,_| k == project_media.id }]
     end
 
+    def safe_get_sync(project_media, field, threshold)
+      response = get_sync(project_media, field, threshold)
+      retries = 0
+      while (response == nil or response["result"] == nil) and retries < 3
```

Comment:
Suggestion for better readability: Use `nil?` instead of ` == nil`, like this:
```
while (response.nil? || response["result"].nil?) && retries < 3
```